### ISSUE TITLE
PyInstaller DLL load failed: pywrap_tensorflow_internal Tensorflow 2.2

### ISSUE THREAD

**Issue by tuvolon** [pyinstaller/pyinstaller#4969]
Tensorflow 2.2 doesn't work with Pyinstaller. I have Pyinstaller 4.0-dev and Tensorflow 2.2.0, and I keep getting the error: 
ImportError: No module named '_pywrap_tensorflow_internal'.
ImportError:DLL load failed: The specified module could not be found.

Failed to load the native Tensorflow runtime.

**Comment by tuvolon** [comment]
@Legorooj 
It seems the problem happens on computer which don't have Microsoft Visual C++ Redistributable 2015-2019 installed, so it can't find the msvcp140(_1).dll files. I think that the individual tensorflow init and self check files need to be changed so that it points to the new msvcp file locations in client computers. Is there any way to fix this in Pyinstaller?

**Comment by Legorooj** [comment]
@tuvolon yes. You need to write a runtime hook. I've got one I need to write today anyway so I'll link to an example of a similar one to the one TF will need later.

**Comment by tuvolon** [comment]
Sure thing. What should the hook include? I'm looking through all the files that are required, and it seems that there needs to be more than just MSVCP140 and MSVCP140_1. I have both of those files, but it won't load. Only when I install Visual C++ Redistributable 2015 - 2019 does it end up working.

**Comment by Legorooj** [comment]
@tuvolon PyInstaller should include those DLLs just fine. We probably just need to tell TF where to find said DLLs, though it should work out of the box

**Comment by tuvolon** [comment]
It's finding the MSVCP140.DLL and MSVCP140_1.DLL files when they are in the app folder, not just in system32.

**Comment by tuvolon** [comment]
Figured it out. The missing DLLs are the files from MS V C++ Redist 2015-2019.
concrt140.dll
mfc140.dll
mfc140chs.dll
mfc140cht.dll
mfc140deu.dll
mfc140enu.dll
mfc140esn.dll
mfc140fra.dll
mfc140ita.dll
mfc140jpn.dll
mfc140kor.dll
mfc140rus.dll
mfc140u.dll
mfcm140.dll
mfcm140u.dll
msvcp140.dll
msvcp140_1.dll
msvcp140_2.dll
msvcp140_codecvt_ids.dll
vcamp140.dll
vccorlib140.dll
vcomp140.dll
vcruntime140.dll
vcruntime140_1.dll
I imported them manually into the app folder and it worked.
